### PR TITLE
Add support for Views DDL

### PR DESCRIPTION
**Pull request by frouioui**
## Description

This PR adds DDL support for Views. The views will be stored by vttablet's underlying storage layer (MySQL) inside internal `_vt` schema in the `views` table.

## Related Issue(s)

- Part of https://github.com/vitessio/vitess/issues/11559

## Checklist

-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

**Review by systay — CHANGES_REQUESTED** [review]
Instead of sending down a query, I think we want to use an RPC to the tablet

**Comment by harshit-gangal** [comment]
All CI checks are failing as I had a `skip CI` label on the PR. It will run on the next commit push.

File: go/vt/vttablet/tabletserver/query_executor.go [hunk 1]
```diff
@@ -289,6 +293,130 @@ func (qre *QueryExecutor) txConnExec(conn *StatefulConnection) (*sqltypes.Result
...
+		Select:      stmt.Select,
+		CheckOption: stmt.CheckOption,
+		Comments:    stmt.Comments,
+	}
+	bindVars := qre.generateBindVarsForViewDDLInsert(createViewDDL)
+	sql, _, err := qre.generateFinalSQL(qre.plan.FullQuery, bindVars)
+	if err != nil {
+		return nil, err
+	}
+	qr, err := withddl.New([]string{mysql.CreateViewsTable}).Exec(qre.ctx, sql, conn.Exec, conn.Exec)
```
[the remark this answers is not in the document]
Comment (harshit-gangal) [reply go/vt/vttablet/tabletserver/query_executor.go:343]: **ack**

**Comment by harshit-gangal** [comment]
> Looks pretty good. Had one small comment, and one question.
> 
> Should we not add the enable-views flag to the `vtexplain` binary as well?

I do not think we can support views on vtexplain. As there is no storage layer and/or communication layer (health checks).

File: go/vt/vttablet/tabletserver/query_executor.go [hunk 2]
```diff
@@ -289,6 +293,130 @@ func (qre *QueryExecutor) txConnExec(conn *StatefulConnection) (*sqltypes.Result
...
+	}
+	sql, _, err := qre.generateFinalSQL(p.GenerateFullQuery(sel), bindVars)
+	if err != nil {
+		return err
+	}
+
+	qr, err := withddl.New([]string{mysql.CreateViewsTable}).Exec(qre.ctx, sql, conn.Exec, conn.Exec)
+	if err != nil {
+		return err
+	}
```
Comment (GuptaManan100) [inline go/vt/vttablet/tabletserver/query_executor.go:405]: Do we really need it to be withDDL, with the create view query given that we already add it to the inializaiton?

**Review by GuptaManan100 — APPROVED** [review]
Same comment for other usages withDDL, otherwise LGTM

**Comment by GuptaManan100** [comment]
@systay with the work we are doing to add `vexplain` command, won't it make sense to just deprecate the `vtexplain` binary entirely? We can support the views very easily in `vexplain` from vtgate, with almost no effort.

**Comment by systay** [comment]
> @systay with the work we are doing to add `vexplain` command, won't it make sense to just deprecate the `vtexplain` binary entirely? We can support the views very easily in `vexplain` from vtgate, with almost no effort.

No-one wants to get rid of vtexplain more than me, but `vexplain` is not enough. One very common usage pattern for vtexplain is to check how queries would behave in sharded mode given different vschemas, and that is difficult to do using `vexplain`

**Comment by GuptaManan100** [comment]
I see. You make a very good point.

File: go/vt/vttablet/tabletserver/query_executor.go (hunk 2 above)
Comment (harshit-gangal) [reply go/vt/vttablet/tabletserver/query_executor.go:405]: View table gets created when the first time view DDL is received. 
If we do not use with, then have to check on the error message received whether it is due to non-existing _vt.views or something else.

Comment (harshit-gangal) [reply go/vt/vttablet/tabletserver/query_executor.go:405]: the initialization is in the health_streamer and we have seen in the past that it sometimes is behind. In the meantime a query can be received which will return with _vt.Views not existing.

**Review by systay — APPROVED** [review]

**Pull request merged**